docs/proposals/propagation.md

```diff
@@ -0,0 +1,142 @@
+# HostPath Volume Propagation
+
+## Abstract
+
+A proposal to add support for propagation mode in HostPath volume, which allows
+mounts within containers to visible outside the container and mounts after pods
+creation visible to containers.
+
+Several existing issues and PRs were already created regarding that particular
+subject:
+* Capability to specify mount propagation mode of per volume with docker [#20698] (https://github.com/kubernetes/kubernetes/pull/20698)
+* Set propagation to "shared" for hostPath volume [#31504] (https://github.com/kubernetes/kubernetes/pull/31504)
+
+## Use Cases
+
+1. (From @Kaffa-MY) Our team attempts to containerize flocker with zfs as back-end
+storage, and launch them in DaemonSet. Containers in the same flocker node need
+to read/write and share the same mounted volume. Currently the volume mount
+propagation mode cannot be specified between the host and the container, and then
+the volume mount of each container would be isolated from each other.
+This use case is also referenced by Containerized Volume Client Drivers - Design
+Proposal [#22216] (https://github.com/kubernetes/kubernetes/pull/22216)
+
+1. (From @majewsky) I'm currently putting the [OpenStack Swift object storage] (https://github.com/openstack/swift) into
+k8s on CoreOS. Swift's storage services expect storage drives to be mounted at
+/srv/node/{drive-id} (where {drive-id} is defined by the cluster's ring, the topology
+description data structure which is shared between all cluster members). Because
+there are several such services on each node (about a dozen, actually), I assemble
+/srv/node in the host mount namespace, and pass it into the containers as a hostPath
+volume.
+Swift is designed such that drives can be mounted and unmounted at any time (most
+importantly to hot-swap failed drives) and the services can keep running, but if
+the services run in a slave mount namespace, they won't see the mounts/unmounts
+performed on the host mount namespace until the containers are restarted.
+The shared mount namespace is the correct solution for this AFAICS. Until this
+becomes available in k8s, we will have to have operations restart containers manually
+based on monitoring alerts.
```

Comment:
Later down the thread in #31504, I was told that I only need a slave mount rather than a shared mount (with the default being private rather than slave), so you can do `sed 's/slave/private/g;s/shared/slave/g` in this paragraph.